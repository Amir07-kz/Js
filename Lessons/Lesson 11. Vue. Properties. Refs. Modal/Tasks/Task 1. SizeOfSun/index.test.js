/**
 * @jest-environment jsdom
 */

const { mount } = require("@vue/test-utils");
const TheSun = require("./TheSun.vue");

describe("c.11.1 TheSun", () => {
    const wrapper = mount(TheSun.default);
    expect(wrapper.element).toMatchSnapshot();

    it("c.11.1.1 TheSun has data", () => {
        expect(typeof TheSun.default.data).toBe("function");
    });

    it("c.11.1.2 TheSun is a Vue instance", () => {
        expect(wrapper.vm).toBeTruthy();
    });

    it("c.11.1.3 click on the sun once", async () => {
        const sunImg = wrapper.find('.sun-container__the-sun');
        await sunImg.trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });

    it("c.11.1.4 click on the two more times", async () => {
        const sunImg = wrapper.find('.sun-container__the-sun');
        await sunImg.trigger("click");
        await sunImg.trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });

    it("c.11.1.5 click on the sun one more time", async () => {
        const sunImg = wrapper.find('.sun-container__the-sun');
        await sunImg.trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });
});
