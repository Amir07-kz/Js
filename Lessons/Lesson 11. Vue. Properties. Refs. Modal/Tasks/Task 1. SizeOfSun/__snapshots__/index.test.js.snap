// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  class="sun-container"
>
  <div
    class="sun-container__block sun-container__block--sun"
  >
    <h3>
      Нажми на солнце
    </h3>
     
    <div
      class="sun-container__img"
    />
  </div>
   
  <div
    class="sun-container__block"
  />
</div>
`;

exports[`c.11.1 TheSun c.11.1.3 click on the sun once 1`] = `
<div
  class="sun-container"
>
  <div
    class="sun-container__block sun-container__block--sun"
  >
    <h3>
      Нажми на солнце
    </h3>
     
    <div
      class="sun-container__img"
    >
      <img
        class="sun-container__the-sun"
        src="/Lessons/Lesson 11. Vue. Properties. Refs. Modal/Tasks/Task 1. SizeOfSun/assets/sun.svg"
        style="width: 428px;"
      />
    </div>
  </div>
   
  <div
    class="sun-container__block"
  >
    <h3>
      Год:
    </h3>
     
    <span>
      
            502021
        
    </span>
     
    <h3>
      Диаметр солнца:
    </h3>
     
    <span>
      
            1323065 км
        
    </span>
  </div>
</div>
`;

exports[`c.11.1 TheSun c.11.1.4 click on the two more times 1`] = `
<div
  class="sun-container"
>
  <div
    class="sun-container__block sun-container__block--sun"
  >
    <h3>
      Нажми на солнце
    </h3>
     
    <div
      class="sun-container__img"
    >
      <img
        class="sun-container__the-sun"
        src="/Lessons/Lesson 11. Vue. Properties. Refs. Modal/Tasks/Task 1. SizeOfSun/assets/sun.svg"
        style="width: 387px;"
      />
    </div>
  </div>
   
  <div
    class="sun-container__block"
  >
    <h3>
      Год:
    </h3>
     
    <span>
      
            1502021
        
    </span>
     
    <h3>
      Диаметр солнца:
    </h3>
     
    <span>
      
            1194066 км
        
    </span>
  </div>
</div>
`;

exports[`c.11.1 TheSun c.11.1.5 click on the sun one more time 1`] = `
<div
  class="sun-container"
>
  <div
    class="sun-container__block sun-container__block--sun"
  >
    <h3>
      Нажми на солнце
    </h3>
     
    <div
      class="sun-container__img"
    >
      <img
        class="sun-container__the-sun"
        src="/Lessons/Lesson 11. Vue. Properties. Refs. Modal/Tasks/Task 1. SizeOfSun/assets/sun.svg"
        style="width: 368px;"
      />
    </div>
  </div>
   
  <div
    class="sun-container__block"
  >
    <h3>
      Год:
    </h3>
     
    <span>
      
            2002021
        
    </span>
     
    <h3>
      Диаметр солнца:
    </h3>
     
    <span>
      
            1134363 км
        
    </span>
  </div>
</div>
`;
